mount/mount_darwin.go

```diff
@@ -0,0 +1,118 @@
+/*
+   Copyright The containerd Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package mount
+
+import (
+	"io"
+	"os"
+	"time"
+
+	"github.com/pkg/errors"
+	"golang.org/x/sys/unix"
+)
+
+var (
+	// ErrNotImplementOnUnix is returned for methods that are not implemented
+	ErrNotImplementOnUnix = errors.New("not implemented under darwin")
+)
+
+// Mount to the provided target path.
+//
+// Use symlink instead of union mount on darwin
+func (m *Mount) Mount(target string) error {
```

Comment:
As like Umount, can you handle non-bind mount here as well?

Reply:
I'm hesitant to handle the non-bind mount case in this `Mount` call as currently I have no cases to call with that.  OTOH, `Unmount` is called without a filesystem type so, it is safe to check as you suggested (and I updated).

we already have the following check, which I think it is enough for the moment.

```
if m.Type != "bind" {
	return errors.Errorf("invalid mount type: '%s'", m.Type)
}
```